include/dlaf/matrix/copy.h

```diff
@@ -13,37 +13,58 @@
 #include <pika/execution.hpp>
 #include <pika/future.hpp>
 
+#include "dlaf/common/range2d.h"
 #include "dlaf/matrix/copy_tile.h"
+#include "dlaf/matrix/matrix.h"
 #include "dlaf/types.h"
-#include "dlaf/util_matrix.h"
 
-namespace dlaf {
-namespace matrix {
+namespace dlaf::matrix {
 
-/// Copy values from another matrix.
+/// Copy subset of local tiles from @p source in the range @p idx_source_begin, @p sz to local subset of
+/// tiles in @p dest in the range @p idx_dest_begin, @p sz.
 ///
-/// Given a matrix with the same geometries and distribution, this function submits tasks that will
-/// perform the copy of each tile.
 template <class T, Device Source, Device Destination>
-void copy(Matrix<const T, Source>& source, Matrix<T, Destination>& dest) {
-  const auto& distribution = source.distribution();
-
-  DLAF_ASSERT(matrix::equal_size(source, dest), source, dest);
-  DLAF_ASSERT(matrix::equal_blocksize(source, dest), source, dest);
-  DLAF_ASSERT(matrix::equal_distributions(source, dest), source, dest);
+void copy(LocalTileSize sz, LocalTileIndex idx_source_begin, Matrix<const T, Source>& source,
+          LocalTileIndex idx_dest_begin, Matrix<T, Destination>& dest) {
+  // Note: the constexpr check is ncessary to avoid compiler errors for comparison of distinct pointer types
```

Comment:
Should the comment be rather about why we _can_ (instead of _must_) use constexpr there: i.e. if `Source != Destination` there's no way for the matrices to be the same? To me the current comment sounds like we're avoiding a compilation error for some case where we would actually want to check for equality, but it's not necessary to check for equality in the other cases.

I'm also a bit uneasy about `&source == &dest` since it comes with the implicit assumption that `Matrix` is non-copyable. However, I know we discussed `operator==` for `Matrix` (or `Tile`?) before and it's not obvious what type of equality it should return.

No changes required here, just wanted to highlight this.

Reply:
I changed the comment here, I hope this one is clearer : https://github.com/eth-cscs/DLA-Future/pull/732/commits/45606d4c3fcdc27a7c69a2b3b9ae22a3de9ad17a

Reply:
Looks good to me, thanks!